### PR TITLE
Adding credential description of auto decrypt keys

### LINKED ISSUE #65: [Feature Request] Support a credential description for auto decrypt keys
Be able to support the following configuration:

```json
            {
                "TokenDecryptionCredentials": [
                {
                    "SourceType": "AutoDecryptKeys",
                    "DecryptKeysApplicationTenant": "mytenant.onmicrosoftonline.com"
                    "DecryptKeysProtocol" : "bearer"
                }]
            }
```

This credential does not affect the container or ReferenceOrValue properties which should not be set and should return null for the soruce type "AutoDecryptKeys".

### PR DESCRIPTION
**Pull request by jmprieur**
Fixes #65 

cc:  @jennyf19  @brentschmaltz @RojaEnnam @TimHannMSFT

### CODE REVIEW

File: src/Microsoft.Identity.Abstractions/ApplicationOptions/CredentialDescription.cs [hunk 1]
```diff
@@ -136,6 +136,22 @@ public class CredentialDescription
...
+        /// Tenant in which the application is registered. This property is only used when
+        /// <see cref="SourceType"/> is <see cref="CredentialSource.AutoDecryptKeys"/>, in order
+        /// to determine the authority to use to get a token to get the decrypt the keys. The cloud
+        /// instance will be the same as the application, but the application can be a multi-tenant
+        /// application (tenant = <b>common</b> or <b>organizations</b>), and in this case to get a token, the
+        /// credential type needs to provide a tenant.
+        /// </summary>
+        public string? DecryptKeysApplicationTenant { get; set; }
+
+        /// <summary>
```
Comment (TimHannMSFT) [inline src/Microsoft.Identity.Abstractions/ApplicationOptions/CredentialDescription.cs:149]: Mention in doc this defaults to bearer?

**Review by TimHannMSFT — APPROVED** [review]

File: src/Microsoft.Identity.Abstractions/ApplicationOptions/CredentialDescription.cs (hunk 1 above)
Comment (TimHannMSFT) [reply src/Microsoft.Identity.Abstractions/ApplicationOptions/CredentialDescription.cs:149]: Hmm this comment popped up above where I meant it to. This is in reference to DecryptKeysProtocol property header.

Comment (jmprieur) [reply src/Microsoft.Identity.Abstractions/ApplicationOptions/CredentialDescription.cs:149]: > Mention in doc this defaults to bearer?

Addressed (in the right comment, lol!)

**Review by RojaEnnam — APPROVED** [review]

**Comment by jmprieur** [comment]
Closing as superseeded by https://github.com/AzureAD/microsoft-identity-abstractions-for-dotnet/pull/66

**Pull request closed without merging**